### PR TITLE
Set up testing with the standard libraries coming from the OS

### PR DESCRIPTION
**Pull request by shajrawi**
We  default Swift tests to use the just-built libraries
See radars rdar://problem/35163663 and rdar://problem/42176864

### CODE REVIEW

**Comment by shajrawi** [comment]
@shahmishal Can you please review?

**Comment by shajrawi** [comment]
@swift-ci Please test

**Comment by shajrawi** [comment]
@swift-ci Please test

**Comment by shajrawi** [comment]
@swift-ci Please test

**Comment by shajrawi** [comment]
@swift-ci Please test

**Comment by shajrawi** [comment]
@swift-ci Please test

File: test/lit.cfg [hunk 1]
```diff
@@ -1461,6 +1461,29 @@ if os.path.exists(libswiftCore_path):
     config.substitutions.append(('%target-static-stdlib-path', static_stdlib_path))
     lit_config.note('using static stdlib path: %s' % static_stdlib_path)
 
+# Set up testing with the standard libraries coming from the OS / just-built libraries
+# default Swift tests to use the just-built libraries
+# For remote runs, this is already supported in REMOTE_RUN_CHILD_DYLD_LIBRARY_PATH above
```
Comment (jrose-apple) [inline test/lit.cfg]: I don't quite get this part. For remote runs, you'd have to *not* provide REMOTE_RUN_CHILD_DYLD_LIBRARY_PATH to get it to use the OS libraries, right?

Comment (shajrawi) [reply test/lit.cfg]: For remote runs, `REMOTE_RUN_CHILD_DYLD_LIBRARY_PATH` is all that matters - you can choose between the OS libraries and the just-built libraries by passing a different `remote_tmp_dir` parameter - you get to choose between the two options today. this new PR is for supporting said choice on a local target

**Comment by shajrawi** [comment]
Changed the PR to have the just-built libraries in the `DYLD_LIBRARY_PATH` even when running with OS libraries based on @shahmishal 's review

**Comment by shajrawi** [comment]
@swift-ci Please smoke test and merge

**Comment by shajrawi** [comment]
@swift-ci Please smoke test and merge

File: test/lit.cfg (hunk 1 above)
Comment (jrose-apple) [reply test/lit.cfg]: That's not correct. `remote_tmp_dir` is for uploads; you can't set it to /usr/lib/swift and have things work out. You'll need to honor this new option there too.

Comment (shajrawi) [reply test/lit.cfg]: ah, you're right, updated the PR accordingly.

**Comment by shajrawi** [comment]
@swift-ci Please smoke test

**Comment by shajrawi** [comment]
@swift-ci Please smoke test

File: test/lit.cfg [hunk 2]
```diff
@@ -1093,6 +1093,15 @@ if (not getattr(config, 'target_run', None) and
     remote_run_host = lit_config.params['remote_run_host']
     remote_tmp_dir = lit_config.params['remote_run_tmpdir']
     remote_lib_dir = os.path.join(remote_tmp_dir, 'stdlib')
+    remote_run_lib_path = ''
+    if 'use_os_stdlib' not in lit_config.params:
+        remote_run_lib_path = remote_lib_dir
+    else:
+		os_stdlib_path = ''
```
Comment (jrose-apple) [inline test/lit.cfg]: Some tabs snuck in here.

**Review by jrose-apple — COMMENTED** [review]
Thanks!

File: test/lit.cfg (hunk 2 above)
Comment (shajrawi) [reply test/lit.cfg]: whoops - accidentally missed the branch up when trying to fix the tab - sorry - will re-open a PR

Comment (shajrawi) [reply test/lit.cfg]: https://github.com/apple/swift/pull/23599

**Pull request closed without merging**